### PR TITLE
adds bioterror foam grenades

### PR DESCRIPTION
**Pull request by LanCartwright**
ignore this part I was drunk

This adds foam grenades for 5 tc each that are basically the equivalent of an EMP grenade....

BUT TO USE AS CROWD CONTROL AGAINST CARBON LIFE FORMS.

THROW ONE AT MEDBAY AND SEE WHOS THE BOSS OFF THE GYM

### CODE REVIEW

**Comment by Iamgoofball** [comment]
You should really make them use the bioterror mix in the sprayer.

**Comment by swankcookie** [comment]
oh I like this :+1:

**Comment by LanCartwright** [comment]
too strong also last I used it I blinded myself

this mix is interesting though, since you can walk into a crowd holding this and they can't say shit about it literally.

**Comment by Incoming5643** [comment]
tex1_32x40_5ee625aafcc25054_5

**Comment by LanCartwright** [comment]
jesus christ goof the khems in that is not kosher


i have to review that stuff man see if it passes halal standards.

**Comment by swankcookie** [comment]
goddamn, just tested it.  that range is fucked up, you have to toss, then run AFAP the other direction

**Comment by swankcookie** [comment]
@LanCartwright onething I would add: sacrifce a chemical for a virus, so that people who have been effected by it can still spread the effect

**Comment by Shapsy** [comment]
Don't ops already have tear gas and flashbangs?

**Comment by LanCartwright** [comment]
yeah but they suck

will consider mmaking new virus nukeop only.

like wizarditis but better

File: code/game/objects/items/weapons/grenades/chem_grenade.dm [hunk 1]
```diff
@@ -421,6 +421,26 @@
...
+	var/obj/item/weapon/reagent_containers/glass/beaker/bluespace/B2 = new(src)
+
+	B1.reagents.add_reagent("cryptobiolin", 75)
+	B1.reagents.add_reagent("potassium", 50)
+	B1.reagents.add_reagent("mutetoxin", 50)
+	B1.reagents.add_reagent("spore", 75)
+	B1.reagents.add_reagent("itching_powder", 50)
+	B2.reagents.add_reagent("phosphorus", 50)
+	B2.reagents.add_reagent("sugar", 50)
+	B2.reagents.add_reagent("mutagen", 200)
```
Comment (PKPenguin321) [inline code/game/objects/items/weapons/grenades/chem_grenade.dm]: holy shit

Comment (swankcookie) [inline code/game/objects/items/weapons/grenades/chem_grenade.dm]: yeah. even if your in it for like 1 tick, you get confused, so its harder too get away.  this is some sarin-level shit.

**Comment by Supermichael777** [comment]
I guarantee the ops will manage to tpk themselves first round this is in.

File: code/game/objects/items/weapons/grenades/chem_grenade.dm (hunk 1 above)
Comment (LanCartwright) [inline code/game/objects/items/weapons/grenades/chem_grenade.dm]: You do not wanna see sarin then.

**Comment by Zilenan91** [comment]
How about giving Ops a virus that makes people vomit up spiders.

**Comment by LanCartwright** [comment]
Sounds too much of a changeling virus. May consider.
Fun idea for a virus: mind control. If infected, the virus will eventually take over the players mind (someone else in control now) and gives you an objective to do X.

Of course, some smartass engineered these and the activity of the virus to match the ideals of the syndicate.

**Comment by Lumipharon** [comment]
Fuck yeah.
From the sounds of it, the range might be a tad much though. If you let that off in maint or some shit... oh lawd, the range.

**Comment by KorPhaeron** [comment]
>Hold grenade in hand
>run towards security
>they all die in seconds while you are immune

Why do ops need an aoe instant death weapon that requires no risk to use like this?

**Comment by Iamgoofball** [comment]
make it foam instead

problem solved

**Comment by LanCartwright** [comment]
@KorPhaeron howe 2 counter biohazard sprays
step 1 put sealed internals
step 2 activate oxygen
step 3 return fire


now if you're a hardcore *baller* you can wear a biosuit (woah squats) and a sterile mask

or powergame like crazy and wear a spacesuit.

**Comment by Zilenan91** [comment]
Make it foam. I doubt sec officers are gonna run around, or even be able to access their suits when they need them.

**Comment by LanCartwright** [comment]
done its now foam your funeral

**Comment by Lumipharon** [comment]
>went from smoke insta rape to foam 10 year creeping tide of terror

Hm.

File: code/datums/uplink_item.dm [hunk 2]
```diff
@@ -219,6 +219,14 @@ var/list/uplink_items = list()
 	gamemodes = list(/datum/game_mode/nuclear)
 	surplus = 35
 
+/datum/uplink_item/dangerous/bioterrorfoam
+	name = "Crowd control chemical foam grenade"
+	desc = "A powerful chemical foam grenade which creates a deadly torrent of foam that will mute, blind, confuse, mutate, and irritate carbon lifeforms. Specially brewed by Tiger Cooperative chemical weapons specialists using additional spore toxin. Ensure suit is sealed before use."
+	item = /obj/item/weapon/grenade/chem_grenade/bioterrorfoam
+	cost = 5
+	gamemodes = list(/datum/game_mode/nuclear,/datum/game_mode/gang)
```
Comment (tkdrg) [inline code/datums/uplink_item.dm]: Why gang?

File: code/game/objects/items/weapons/grenades/chem_grenade.dm [hunk 3]
```diff
@@ -421,6 +421,25 @@
 	beakers += B1
 	beakers += B2
 
+/obj/item/weapon/grenade/chem_grenade/bioterrorfoam
+	name = "Bio terror foam grenade"
+	desc = "Tiger Cooperative chemical foam grenade. Causes temporary irration, blindness, confusion, autism, and mutations to carbon based life forms. Contains additional spore toxin"
```
Comment (tkdrg) [inline code/game/objects/items/weapons/grenades/chem_grenade.dm]: > autism
that was a joke to be quite honest my friend
"loss of speech" or something like that
also you're missing the full stop here

File: code/datums/uplink_item.dm (hunk 2 above)
Comment (LanCartwright) [inline code/datums/uplink_item.dm]: I was definitely drunk when I decided to add it to hang. But I think the reasoning was something along the lines of "keep people away from the dominator" and "get people away from their dominator" kind of scenario.

In other words, purely crowd control.

**Comment by tkdrg** [comment]
Could you also squash these commits, you don't need 12+ commits for this

File: code/game/objects/items/weapons/grenades/chem_grenade.dm (hunk 3 above)
Comment (LanCartwright) [inline code/game/objects/items/weapons/grenades/chem_grenade.dm]: Yeah Antur told me to change mutism to autism as a joke.

File: code/datums/uplink_item.dm (hunk 2 above)
Comment (tkdrg) [inline code/datums/uplink_item.dm]: I dunno about gang, I think it doesn't really fit in with the style and stuff.

Comment (LanCartwright) [inline code/datums/uplink_item.dm]: Alright I'll rip it out of gang tomorrow. On hindsight it didn't fit and was probably op as fuck.

**Comment by LanCartwright** [comment]
Don't know how to :|
Should I close + new branch + pull it?

**Comment by tkdrg** [comment]
Nope. Just `git rebase -i HEAD~13`, mark you first commit as `pick` and the rest as `squash`. You can always ask #coderbus for help as well

File: code/game/objects/items/weapons/grenades/chem_grenade.dm (hunk 3 above)
Comment (Dorsisdwarf) [inline code/game/objects/items/weapons/grenades/chem_grenade.dm]: the joke is the targets all already have it because they're playing spacemens

**Pull request closed without merging**